CMakeLists.txt

```diff
@@ -93,6 +93,12 @@ check_cxx_source_compiles ("#include <sys/types.h>
   " XATTR_ADD_OPT)
 set (CMAKE_REQUIRED_QUIET ${CMAKE_REQUIRED_QUIET_SAVE})
 
+# Check if we have fdatasync.
+include (CheckCXXSourceCompiles)
```

Comment:
Of course if it works, let's go with this nicer solution 👍 
Was not aware of it (well, did not look after it :)